### PR TITLE
Tests: Support for new cucumber app call txn decoding test

### PR DESCRIPTION
**Pull request by jasonpaulos**
This PR adds support for https://github.com/algorand/algorand-sdk-testing/pull/248

Note: the code exported by this library does not suffer from the same bug as https://github.com/algorand/js-algorand-sdk/issues/688, but some changes were needed to get the test to pass properly.

I don't believe this PR necessitates a patch release.

### CODE REVIEW

File: .test-env
Comment (jasonpaulos) [inline .test-env:3]: Reminder to revert
```suggestion
SDK_TESTING_BRANCH="master"
```

File: test/steps_test.go
Comment (jasonpaulos) [inline test/steps_test.go:380]: This step was registered twice, the other time being https://github.com/algorand/go-algorand-sdk/blob/60292fb770fcd8f94f881cbf7b117affc24a6086/test/transactions_test.go#L195

Comment (algochoi) [reply test/steps_test.go:380]: Good catch

File: test/steps_test.go
Comment (algochoi) [inline test/steps_test.go:2507]: nit  - spelling preference : 
```suggestion
	// This test isn't perfect as it's sensitive to non-meaningful changes (e.g. nil slice vs 0
```

**Review by algochoi — APPROVED** [review]
Changes look good to merge after merging the upstream SDK Testing PR - looks like tests were successfully run in CI.

File: test/steps_test.go
Comment (jasonpaulos) [reply test/steps_test.go]: Thanks!

**Pull request merged**
